client/src/components/Grid/configs/historiesShared.ts

```diff
@@ -0,0 +1,120 @@
+import { faEye } from "@fortawesome/free-solid-svg-icons";
+import { useEventBus } from "@vueuse/core";
+
+import { historiesQuery } from "@/api/histories";
+import { updateTags } from "@/api/tags";
+import Filtering, { contains, expandNameTag, type ValidFilter } from "@/utils/filtering";
+import _l from "@/utils/localization";
+import { rethrowSimple } from "@/utils/simple-error";
+
+import type { FieldArray, GridConfig } from "./types";
+
+const { emit } = useEventBus<string>("grid-router-push");
+
+/**
+ * Local types
+ */
+type HistoryEntry = Record<string, unknown>;
+type SortKeyLiteral = "create_time" | "name" | "update_time" | undefined;
+
+/**
+ * Request and return data from server
+ */
+async function getData(offset: number, limit: number, search: string, sort_by: string, sort_desc: boolean) {
+    const { data, headers } = await historiesQuery({
+        limit,
+        offset,
+        search,
+        sort_by: sort_by as SortKeyLiteral,
+        sort_desc,
+        show_own: false,
+        show_published: false,
+        show_shared: true,
+    });
+    const totalMatches = parseInt(headers.get("total_matches") ?? "0");
+    return [data, totalMatches];
+}
+
+/**
+ * Declare columns to be displayed
+ */
+const fields: FieldArray = [
+    {
+        key: "name",
+        title: "Name",
+        type: "operations",
+        operations: [
+            {
+                title: "View",
+                icon: faEye,
+                condition: (data: HistoryEntry) => !data.deleted,
+                handler: (data: HistoryEntry) => {
+                    emit(`/histories/view?id=${data.id}`);
+                },
+            },
+        ],
+    },
+    {
+        key: "id",
+        title: "Size",
+        type: "datasets",
+    },
+    {
+        key: "tags",
+        title: "Tags",
+        type: "tags",
+        handler: async (data: HistoryEntry) => {
```

Comment:
Users can not change shared history tags that are not owned by them:
```suggestion
        disabled: true,
        handler: async (data: HistoryEntry) => {
```